recipe-server/client/control/components/Fields.js

```diff
@@ -50,34 +50,43 @@ buildControlField.propTypes = {
   children: pt.node,
 };
 
+export const CheckboxGroup = ({ name, onChange, options = [], value }) => {
+  /**
+   * Checkbox change event handler. Appends or removes the selected checkbox's
+   * value to the existing `value` prop, and reports the change up to redux-form.
+   *
+   * @param  {Event} onChange event object
+   */
+  const handleChange = ({ event: target }) => {
```

Comment:
> Using onChange={handleChange}, where handleChange is defined in CheckboxGroup, would loop back around to the issue that handleChange is defined every render, which re-renders the subcomponents.

Which means `handleChange` should also be a method on a component class instead of a within a function. I think the general rule is that anonymous functions should be avoided in components, wherever they are.

> Furthermore, since these 'components' are not actual React Components, I'm fairly positive that the whole list is rendered regardless of what prop changes are passed in.

`render` (or functional components) always gets called, so there's no avoiding that. What's avoided is messing with the DOM, which is generally the much-slower part. Functional components like this aren't different from class-based components in that regard.

> Perhaps we should actually just use React.Component and define handleChange as a component function, which would avoid all of this. This code should be sufficient.

Using the value on the input is another fine way of doing this. I think I prefer it to my own suggestion, actually. 👍